pagopa/pn-frontend
# Feature/pn-10866 - Add layer for statistics data transformation (PA)

## Short description
This PR adds the utility/StatisticsData module inside the pa-webapp to achieve the goal of performing data transformation from the dto received from the be to a collection of data structures which are ready to be used by the components which form the sender dashboard.

## List of changes proposed in this pull request
- add StatisticsData abstract class, which act as superclass for every concrete StatisticsData
- add a collection of specialized StatisticsData classes. Any single class implements the logic to generate the transformed data for a particular component to be presented on the sender dashboard:
  - FiledStatisticsData
  - LastStateStatisticsData
  - DeliveryModeStatisticsData
  - DigitalStateStatisticsData
  - DigitalMeanTimeStatisticsData
  - DigitalErrorsDetailStatisticsData
  - DigitalAttemptsStatisticsData
- add StatisticsDataFactory which is responsible for instantiating objects having type StatisticsData and use them to build the transformed data
- add StatisticsDataFactoryManager which is responsible for creating, storing and managing a factory instance

## How to test
Go to Statistics page and verify the statistics slice of the redux store got updated with proper data
Be sure to set the IS_STATISTICS_ENABLED environment variable to true in conf/config.json
